pkg/descheduler/node/node_test.go

```diff
@@ -215,238 +245,527 @@ func TestPodFitsAnyOtherNode(t *testing.T) {
 		pod         *v1.Pod
 		nodes       []*v1.Node
 		success     bool
+		podsOnNodes []*v1.Pod
 	}{
 		{
 			description: "Pod fits another node matching node affinity",
-			pod:         createPodManifest(nodeNames[2], nodeLabelKey, nodeLabelValue),
```

Comment:
The `createPodManifest` is/was setting `Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution` whereas here the test is changed to set `spec.NodeSelector`. Which is not the same as the node affinity. Would you elaborate on why this change was performed?